### ISSUE TITLE
Draggable annotation position wrong with negative x/y

### ISSUE THREAD

**Issue by wilywampa** [matplotlib/matplotlib#4295]
Any draggable annotation is displayed in the wrong position after dragging (i.e. nowhere near the mouse cursor) if either of the initial XY coordinates is negative. Minimal example:
![drag](https://cloud.githubusercontent.com/assets/4472522/6884211/09f0fa00-d59a-11e4-9b2e-fa54be93ebb6.gif)

```python
import matplotlib.pyplot as plt
import matplotlib as mpl
mpl.rcdefaults()
mpl.rcParams['toolbar'] = 'none'

plt.figure(figsize=(5, 2))
plt.plot(range(-10, 10))
an1 = plt.annotate(s='good', xy=(15, 5), bbox={'alpha': 0.5})
an1.draggable()
an2 = plt.annotate(s='bad', xy=(5, -5), bbox={'alpha': 0.5})
an2.draggable()
plt.show()
```
It is caused by the reference coordinates being set apparently incorrectly here: https://github.com/matplotlib/matplotlib/blob/8a270fcc64197ee00c818ec8f9ccc9f0e884828a/lib/matplotlib/text.py#L1597-1605

Using matplotlib 1.4.3 with various OSes/backends.

**Comment by efiring** [comment]
Looks like a duplicate of #4292.  As noted there, and as is evident from the code, the behavior is deliberate but the rationale for it is unclear.

**Comment by tacaswell** [comment]
This in now fixed in 1.5.1+
